src/main/java/com/bookshop/dto/CreateBookRequestDto.java

```diff
@@ -1,12 +1,25 @@
 package com.bookshop.dto;
 
+import com.bookshop.validation.Isbn;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 
 public record CreateBookRequestDto(
+        @NotNull
+        @Size(max = 255)
         String title,
+        @NotNull
+        @Size(max = 255)
         String author,
+        @Isbn
```

Comment:
Just want to mention that you can use @ISBN from org.hibernate.validator.constraints, but I'm sure that you know it without me))

Reply:
I just tried to create my own annotation with some validation logic.